app/src/main/java/dev/arkbuilders/rate/presentation/quick/QuickViewModel.kt

```diff
@@ -72,34 +73,46 @@ class QuickViewModel(
                 postSideEffect(QuickScreenEffect.ShowSnackbarAdded(visuals))
             }.launchIn(viewModelScope)
 
-            quickRepo.allFlow().onEach { all ->
-                val codeToRate = currencyRepo.getCodeToCurrencyRate().getOrNull()!!
-                val displayList = all.reversed().map { pair ->
-                    val toDisplay = pair.to.map { code ->
-                        val (amount, _) = convertUseCase(
-                            Amount(pair.from, pair.amount),
-                            toCode = code,
-                            codeToRate
-                        )
-                        amount
-                    }
-                    QuickDisplayPair(pair, toDisplay)
-                }
-                val pages = displayList.groupBy { it.pair.group }
+            quickRepo.allFlow().drop(1).onEach { quick ->
+                val pages = quick.reversed().groupBy { it.group }
                     .map { (group, pairs) -> QuickScreenPage(group, pairs) }
                 intent {
                     reduce {
                         state.copy(
-                            pages = pages,
-                            initialized = true
+                            pages = pages
                         )
                     }
                 }
             }.launchIn(viewModelScope)
 
-            val names = currencyRepo.getCurrencyName().getOrNull()!!
+            calcFrequentCurrUseCase.flow().drop(1).onEach {
+                val currencies = currencyRepo.getCurrencyNameUnsafe()
+                val frequent = calcFrequentCurrUseCase.invoke()
+                    .map { currencyRepo.nameByCodeUnsafe(it) }
+                val topResults = frequent + (currencies - frequent)
+                reduce {
+                    state.copy(
+                        frequent = frequent,
+                        topResults = topResults
+                    )
+                }
+            }.launchIn(viewModelScope)
+
+            val all = currencyRepo.getCurrencyNameUnsafe()
+            val frequent = calcFrequentCurrUseCase()
+                .map { currencyRepo.nameByCodeUnsafe(it) }
+            val topResults = frequent + (all - frequent)
```

Comment:
Nice trick to move the frequent list to the top. 😂 Perhaps consider having 2 use cases like `GetTopResult` and other use case to encapsulate these logic

Reply:
It's a good idea to extract use case.